src/test/java/com/peel/okhttp3/staleiferror/StaleIfErrorInterceptorTest.java

```diff
@@ -15,14 +15,310 @@
  */
 package com.peel.okhttp3.staleiferror;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.fail;
 
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 
+import okhttp3.Cache;
+import okhttp3.CacheControl;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.tls.HandshakeCertificates;
+import static okhttp3.tls.internal.TlsUtil.localhost;
+
+/**
+ * 
+ * @author swaroop
+ */
 public class StaleIfErrorInterceptorTest {
 
-    @Test
-    public void test() {
-        fail("Not yet implemented");
-    }
+	@Rule
+	public final MockWebServer server = new MockWebServer();
+
+	private HandshakeCertificates handshakeCertificates = localhost();
+
+	private OkHttpClient client;
+	private OkHttpClient clientWith4SecMaxStale;
+
+	private HttpUrl url;
+
+	@Before
+	public void setUp() {
+		client = new OkHttpClient.Builder()
+				.addInterceptor(new StaleIfErrorInterceptor())
+				.sslSocketFactory(handshakeCertificates.sslSocketFactory(), handshakeCertificates.trustManager())
```

Comment:
Do we really need sslSocketFactory?

Reply:
we don't need it for our current tests, removed it.